python/lsst/meas/algorithms/loadReferenceObjects.py

```diff
@@ -369,30 +370,35 @@ def makeMinimalSchema(filterNameList, *, addCentroid=False,
 
             - If 2 or 3: add fields "coord_raErr" and "coord_decErr".
             - If 3: also add field "coord_radecErr".
+
+            Ignored if ``fullPositionCovariance=True``.
         addProperMotion : `bool`
             If True add fields "epoch", "pm_ra", "pm_dec" and "pm_flag".
         properMotionErrDim : `int`
-            Number of proper motion error fields; must be one of 0, 2, 3;
-            ignored if addProperMotion false:
-            - If 2 or 3: add fields "pm_raErr" and "pm_decErr".
-            - If 3: also add field "pm_radecErr".
+            Deprecated (see DM-?????): Use ``fullPositionCovariance`` when proper motions
+            are provided, as they will be correlated with the other position
+            paramters.
         addParallax : `bool`
             If True add fields "epoch", "parallax", "parallaxErr"
             and "parallax_flag".
+        fullPositionCovariance : `bool`
+            Include a full coordinate+PM+parallax covariance matrix. Requires
+            that `addProperMotion` and `addParallax` are set.
+            Ignores the `coordErrDim` and  `properMotionErrDim` parameters.
```

Comment:
You're probably right that `addProperMotion` and `addParallax` should go away, too.

Our logic here is that either you have PM+parallax+all covariance (Gaia), or you don't care about any position errors beyond ra/dec (anything non-Gaia). Gaia is the only relevant astrometric catalog (until we make our own, and we won't need this code for that), and whether or not a given gaia star has the full 5-parameter covariance is irrelevant to whether the refcat should have it (since it needs to be able to hold all the covariances).

Reply:
My point is that you're adding lots of API and maintenance complexity for the sole purpose of reducing functionality compared to if you just did things the simple way.

But even if you are starting from a Gaia catalog (and this is supposed to be the _generic_ converter, not the Gaia specialization), why should users be required to translate parallax in order to get covariances for the other parameters? I can see why LSST needs proper motions, but parallax seems irrelevant to data processing.